lib/mongoid/extensions/regexp.rb

```diff
@@ -12,7 +12,7 @@ module ClassMethods
         # @example Mongoize the object.
         #   Regexp.mongoize(/\A[abc]/)
         #
-        # @param [ Object ] object The object to mongoize.
+        # @param [ Regexp | String ] object The object to mongoize.
```

Comment:
This method actually accepts arbitrary input as of MONGOID-5222

Reply:
Good catch, I've fixed this.